crates/pypi-types/src/metadata/metadata_resolver.rs

```diff
@@ -0,0 +1,229 @@
+//! Derived from `pypi_types_crate`.
+
+use std::str::FromStr;
+
+use itertools::Itertools;
+use serde::{Deserialize, Serialize};
+use tracing::warn;
+
+use pep440_rs::{Version, VersionSpecifiers};
+use pep508_rs::Requirement;
+use uv_normalize::{ExtraName, PackageName};
+
+use crate::lenient_requirement::LenientRequirement;
+use crate::metadata::pyproject_toml::parse_pyproject_toml;
+use crate::metadata::Headers;
+use crate::{metadata, LenientVersionSpecifiers, MetadataError, VerbatimParsedUrl};
+
+/// A subset of the full core metadata specification, including only the
+/// fields that are relevant to dependency resolution.
+///
+/// Core Metadata 2.3 is specified in <https://packaging.python.org/specifications/core-metadata/>.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "kebab-case")]
+pub struct MetadataResolver {
```

Comment:
Why is this one not called Metadata 2.3?

Reply:
I'm adding a `Metadata23` (the full one) in the follow-up PR.